src/client-side-encryption/providers/utils.ts

```diff
@@ -1,16 +1,18 @@
-import { MongoCryptKMSRequestNetworkTimeoutError } from '../errors';
 import * as http from 'http';
+import { clearTimeout, setTimeout } from 'timers';
+
+import { MongoCryptKMSRequestNetworkTimeoutError } from '../errors';
 
 /**
- * @param {URL | string} url
- * @param {http.RequestOptions} options
- *
- * @returns { Promise<{ body: string, status: number }> }
- * @ignore
+ * @internal
  */
-function get(url, options = {}) {
+export function get(
+  url: URL | string,
+  options: http.RequestOptions = {}
+): Promise<{ body: string; status: number | undefined }> {
```

Comment:
This can be deduped with the http request helper in src/utils.ts right?

Reply:
It's not entirely straightforward - you'll notice this method is specialized to throw a `MongoCryptKMSRequestNetworkTimeoutError` error on timeout and to return a parsed JSON body with the status code.  The auth helper throws an auth error and only returns a parsed JSON body.

They can be deduped, but sharing the logic requires changes in both places, no matter which request helper we keep.  I didn't want my changes to spill into auth, so I kept it as-is.

Reply:
_if_ we are going to make this change, I'd suggest we just use fetch (and pull in node_fetch for shell), and then both implementations effectively go away

Reply:
from tech chat, file a follow up